presto-product-tests/src/main/java/io/prestosql/tests/kafka/TestKafkaWritesSmokeTest.java

```diff
@@ -0,0 +1,272 @@
+import static io.prestosql.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static java.lang.String.format;
+
+public class TestKafkaWritesSmokeTest
+        extends ProductTest
+{
+    private static final String KAFKA_CATALOG = "kafka";
+    private static final String SCHEMA_NAME = "product_tests";
+
+    private static final String SIMPLE_KEY_AND_VALUE_TABLE_NAME = "write_simple_key_and_value";
+    private static final String SIMPLE_KEY_AND_VALUE_TOPIC_NAME = "write_simple_key_and_value";
+
+    // kafka-connectors requires tables to be predefined in presto configuration
+    // the requirements here will be used to verify that table actually exists and to
+    // create topics and propagate them with data
+
+    private static class SimpleKeyAndValueTable
+            implements RequirementsProvider
+    {
+        @Override
+        public Requirement getRequirements(Configuration configuration)
+        {
+            return immutableTable(new KafkaTableDefinition(
+                    SCHEMA_NAME + "." + SIMPLE_KEY_AND_VALUE_TABLE_NAME,
+                    SIMPLE_KEY_AND_VALUE_TOPIC_NAME,
+                    new ListKafkaDataSource(ImmutableList.of()),
+                    1,
+                    1));
+        }
+    }
+
+    @Test(groups = {KAFKA, PROFILE_SPECIFIC_TESTS})
+    @Requires(SimpleKeyAndValueTable.class)
+    public void testInsertSimpleKeyAndValue()
+    {
+        assertThat(query(format(
+                "insert into %s.%s.%s values " +
+                        "('jasio', 1, 'ania', 2), " +
+                        "('piotr', 3, 'kasia', 4)",
+                KAFKA_CATALOG,
+                SCHEMA_NAME,
+                SIMPLE_KEY_AND_VALUE_TABLE_NAME
+        ))).updatedRowsCountIsEqualTo(2);
+
+        assertThat(query(format(
+                "select varchar_key, bigint_key, varchar_value, bigint_value from %s.%s.%s",
+                KAFKA_CATALOG,
+                SCHEMA_NAME,
+                SIMPLE_KEY_AND_VALUE_TABLE_NAME
+        ))).containsOnly(
+                row("jasio", 1, "ania", 2),
+                row("piotr", 3, "kasia", 4));
+    }
+
+    private static final String ALL_DATATYPES_RAW_TABLE_NAME = "write_all_datatypes_raw";
+    private static final String ALL_DATATYPES_RAW_TOPIC_NAME = "write_all_datatypes_raw";
+
+    private static class AllDataTypesRawTable
+            implements RequirementsProvider
+    {
+        @Override
+        public Requirement getRequirements(Configuration configuration)
+        {
+            return immutableTable(new KafkaTableDefinition(
+                    SCHEMA_NAME + "." + ALL_DATATYPES_RAW_TABLE_NAME,
+                    ALL_DATATYPES_RAW_TOPIC_NAME,
+                    new ListKafkaDataSource(ImmutableList.of()),
+                    1,
+                    1));
+        }
+    }
+
+    @Test(groups = {KAFKA, PROFILE_SPECIFIC_TESTS})
+    @Requires(AllDataTypesRawTable.class)
+    public void testInsertRawTable()
```

Comment:
do we have a test for inserting nulls? Or is that not supported?

Reply:
Nulls are not supported with Raw encoder/decoder since it encodes/decodes fixed with byte sequences.